source/common/upstream/cluster_manager_impl.cc

```diff
@@ -43,11 +43,22 @@ ClusterManagerImpl::ClusterManagerImpl(const Json::Object& config, Stats::Store&
     loadCluster(cluster, stats, dns_resolver, ssl_context_manager, runtime, random);
   }
 
+  Optional<std::string> local_cluster_name;
+  if (config.hasObject("local_cluster")) {
```

Comment:
I wonder if "local_cluster" might be kind of confusing (though I can't think of a better name). In any case, I think I would change it to "local_cluster_name" and just have it refer to a string name, since ultimately we want it to be a reference into the cluster list.